### PR TITLE
Update nlohmann/json.hpp to version 3.10.5

### PR DESCRIPTION
**Pull request by linev**
Provide json_fwd.h which can be used now in eve7

Also update forward declaration in REveElement.hxx

### CODE REVIEW

**Review by hahnjo — APPROVED** [review]
Looks good. Thanks @linev !

**Review by bellenot — APPROVED** [review]

File: graf3d/eve7/inc/ROOT/REveElement.hxx [hunk 1]
```diff
@@ -30,23 +30,47 @@
...
+         class BinaryType = std::vector<std::uint8_t>>
 class basic_json;
 
-using json = basic_json<std::map, std::vector, std::string, bool, std::int64_t, std::uint64_t, double, std::allocator,
-                        adl_serializer, std::vector<std::uint8_t>>;
-} // namespace nlohmann
+/// @brief JSON Pointer defines a string syntax for identifying a specific value within a JSON document
+/// @sa https://json.nlohmann.me/api/json_pointer/
+template<typename BasicJsonType>
+class json_pointer;
```
Comment (hahnjo) [inline graf3d/eve7/inc/ROOT/REveElement.hxx:60]: We don't need this

File: graf3d/eve7/inc/ROOT/REveElement.hxx [hunk 2]
```diff
@@ -30,23 +30,47 @@
...
+using json = basic_json<>;
+
+/// @brief a minimal map-like container that preserves insertion order
+/// @sa https://json.nlohmann.me/api/ordered_map/
+template<class Key, class T, class IgnoredLess, class Allocator>
+struct ordered_map;
+
+/// @brief specialization that maintains the insertion order of object keys
+/// @sa https://json.nlohmann.me/api/ordered_json/
+using ordered_json = basic_json<nlohmann::ordered_map>;
```
Comment (hahnjo) [inline graf3d/eve7/inc/ROOT/REveElement.hxx:71]: We don't need this

**Review by hahnjo — COMMENTED** [review]
we *can* update our forward declaration, but it's really just a different form of writing it...

File: graf3d/eve7/inc/ROOT/REveElement.hxx (hunk 2 above)
Comment (linev) [reply graf3d/eve7/inc/ROOT/REveElement.hxx:71]: I just copy and paste that is inside `json_fwd.hpp`.
It is not absolutely the same as with v 3.10 - there are default values for the template arguments. 
But anyway soon or later we will remove this part completely.

Comment (hahnjo) [reply graf3d/eve7/inc/ROOT/REveElement.hxx:71]: Okay, your part of the code.

**Comment by hahnjo** [comment]
As described in https://github.com/root-project/root/issues/11236, the current state in `master` causes problems if there is a system installation of `json_fwd.hpp` that is incompatible with the builtin. Can we get this merged to solve the problem? For the same reason, we probably also want a backport to v6.26...

**Pull request merged**